core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -109,6 +109,7 @@ private RpcUtils() {} // prevent instantiation
     // avoid string format for better performance if debug is off
     String debugDesc = logger.isDebugEnabled() ? String.format(description, args) : null;
     try (Timer.Context ctx = MetricsSystem.timer(getQualifiedMetricName(methodName)).time()) {
+      MetricsSystem.counter(getQualifiedInProgressMetricName(methodName)).inc();
```

Comment:
nit: how about moving timer and counter calls together above and close them together on finally below?

Reply:
then we run into situation where ctx might not be initialized before close is called?. i think try-with-resources is the standard way of doing this, and we are not really closing counter right?